api_tests/src/actors/actor.ts

```diff
@@ -470,6 +481,57 @@ export class Actor {
                 );
             }
         }
+
+        this.logger.debug(
+            "%s asset set to %s",
+            this.alphaAsset,
+            this.betaAsset.quantity
+        );
+        this.logger.debug(
+            "%s asset set to %s",
+            this.betaAsset,
+            this.betaAsset.quantity
+        );
+    }
+
+    public async setStartingBalancesForPartialTake(partialTakeAmount: string) {
+        switch (this.name) {
+            case "alice": {
+                // Alice purchases beta asset with alpha asset
+                await this.setStartingBalance([
+                    this.alphaAsset,
+                    {
+                        ...this.betaAsset,
+                        quantity: "0",
+                    },
+                ]);
+                this.expectedBalanceChanges.set(
+                    toKey(this.betaAsset),
+                    BigInt(Number(this.order.rate) * Number(partialTakeAmount))
```

Comment:
Is this math really necessary?

If I am taking only the `partialTakeAmount`, shouldn't my balance change exactly by this amount? :)

Reply:
the partialTakeAmount is specified in bitcoin, the corresponding dai amount for the swap needs to be calculated using by partialTakeAmount * BTCtoDAIConversionRate